src/components/containers/DraggablePanel/DraggablePanel.tsx

```diff
@@ -0,0 +1,196 @@
+  styleOverrides?: DraggablePanelStyleOverrides;
+  /** This event fires when the user's drag completes. */
+  onDragComplete?: (
+    destinationCoordinates: DraggablePanelCoordinatePair
+  ) => void;
+  /** This event fires when the user's drag begins. */
+  onDragStart?: () => void;
+  /** This event fires when the user dismisses a visible panel. */
+  onPanelDismiss?: () => void;
+};
+
+const screenReaderOnly = css`
+  height: 1px;
+  left: -10000px;
+  overflow: hidden;
+  position: absolute;
+  top: auto;
+  width: 1px;
+`;
+
+export function DraggablePanel({
+  confineToParentContainer,
+  children,
+  defaultPosition,
+  initialPanelHeight,
+  initialPanelWidth,
+  isOpen,
+  onDragComplete,
+  onDragStart,
+  onPanelDismiss,
+  panelTitle,
+  showPanelTitle,
+  styleOverrides,
+}: DraggablePanelProps) {
+  const [wasDragged, setWasDragged] = useState<boolean>(false);
+  const [isDragging, setIsDragging] = useState<boolean>(false);
+
+  function handleDrag() {
+    !wasDragged && setWasDragged(true);
+  }
+
+  function handleDragStart() {
+    setIsDragging(true);
+
+    if (onDragStart) onDragStart();
+  }
+
+  function handleOnDragStop(_: DraggableEvent, data: DraggableData) {
+    setIsDragging(false);
+
+    if (onDragComplete) {
+      onDragComplete({
+        x: data.lastX,
+        y: data.lastY,
+      });
+    }
+  }
+
+  return (
+    <Draggable
+      bounds={confineToParentContainer ? "parent" : false}
+      defaultPosition={defaultPosition || { x: 0, y: 0 }}
+      handle=".dragHandle"
+      onDrag={handleDrag}
+      onStart={handleDragStart}
+      onStop={handleOnDragStop}
+    >
+      <div
+        // As the attribute's name suggests, this helps with automated testing.
+        // At the moment, jsdom and dragging is a bad combo for testing.
+        data-testid={`${panelTitle} ${wasDragged ? "dragged" : "not dragged"}`}
+        css={css`
+          background: white;
+          border-radius: 7px;
+          box-shadow: rgba(50, 50, 93, 0.25) 0px 2px 5px -1px,
+            rgba(0, 0, 0, 0.3) 0px 1px 3px -1px;
+          min-width: 250px;
+          position: relative;
+          width: ${initialPanelWidth || "unset"};
+          ${isOpen === false &&
+          `
+          visibility: hidden;
+          `}
+          z-index: ${styleOverrides?.zIndex ?? "auto"}
+        `}
+      >
+        <div
+          className="dragHandle"
+          css={css`
+            align-items: center;
+            border-radius: 7px 7px 0 0;
+            background: ${gray[100]};
+            cursor: ${isDragging ? "grabbing" : "grab"};
+            display: flex;
+            height: 2rem;
+            justify-content: space-between;
+            position: relative;
+            width: 100%;
+          `}
+        >
+          <div
+            aria-hidden
+            css={css`
+              margin-left: 1rem;
+            `}
+          >
+            <DragHandle />
+          </div>
+          <h2
```

Comment:
hi sorry it's me again. I'm looking at the details here as i start to put this in figma :) 
Mind if we use one of CoreUI's heading styles? I'm going to guess H5 is probably the best fit, but see what you think

Reply:
> hi sorry it's me again. I'm looking at the details here as i start to put this in figma :) Mind if we use one of CoreUI's heading styles? I'm going to guess H6 is probably the best fit, but see what you think

Will do! Yeah, I'm planning on use the components native to this repo in addition to doing styling in a way that's coherent with the rest of the repo. I probably shouldn't have made this PR totally ready for review 😆

Reply:
And also, to introduce a rabbit hole I'd like to bring up in future wg-styling meetings: I'm concerned about the typography components. 

Ideally, heading tags are used semantically and not stylistically. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#accessibility_concerns. They should progress sequentially. I do understand that this is difficult in modern, dynamic web applications, but we're still working with a document at the end of the day. I might need to actually use an `<h2>` or `<h3>`, but the our typography components are huge so then I'd have to do a style override.

<img width="514" alt="image" src="https://user-images.githubusercontent.com/24446573/216689489-9f5a7779-b5f8-42d7-96f7-34e8001350fb.png">

Anyways, this is not the venue and I should be ashamed for bringing it up in reply to a PR comment! 🔕

Reply:
Well put. I've been trying to figure out how to articulate my concern about having fixed font sizes for heading elements. I think it will be a tough nut to crack. I don't think reducing the size is going to be enough. For instance, we might typically want `<h2>` elements to be `1.3rem` (eg), but that would be too big in the context of a draggable panel.

Maybe another way to approach it is to decouple styling from semantic element by components like `MainHeading`, `SecondaryHeading`, etc, which can take a component or html element name to use as the container element:

Reply:
oh very interesting. I have a lot of questions but for now i'll just link these comments in the styling wg page! Rest assured that we will tackle this topic asap!!!

Reply:
https://veupathdb.atlassian.net/wiki/spaces/WG/pages/184320076/Revisiting+typography+hierarchy

Reply:
This is great!